### ISSUE TITLE
Replace all mockery mock code with gomock mock code

### ISSUE THREAD

**Issue by qdm12** [ChainSafe/gossamer#2314]
We are now migrating slowly to use gomock which is maintainted by some of the golang team.

Its features are more complete, it is more popular and it has a tighter integration with the Go language.
This would also allow us to use to use Go 1.18 without being blocked by mockery not supporting it as well.

This issue should:

- replace/remove all mockery code with gomock code
- Remove mockery from the documentation in the contributing document
- Remove mockery from the `mocks checks` CI workflow

**Comment by qdm12** [comment]
Duplicate of #1954 sorry about that!
